back/tests/cmd/openbsrv-integration/integration_test.go

```diff
@@ -208,19 +209,14 @@ func userSvcAddAndFndUsersFn(ctx context.Context, conn *grpc.ClientConn, clnt pb
 			got := r.Item
 			wantUserID := got.Id
 
-			// Unset fields that aren't being tested.
-			got.Id = ""
-			got.XXX_unrecognized = []byte{}
-			got.XXX_sizecache = 0
+			unsetUntestedFields(got)
 
 			for i := 0; i < len(got.Roles); i++ {
-				got.Roles[i].Id = ""
-				got.Roles[i].XXX_unrecognized = []byte{}
-				got.Roles[i].XXX_sizecache = 0
+				unsetUntestedFields(got.Roles[i])
 			}
 
 			if reflect.DeepEqual(got, tc.want) {
```

Comment:
Great catch, @daved! I flipped it and I got something unexpected. Why is it that the `got` and `tc.want` are not considered equal?
![Screenshot from 2020-09-21 22-37-46](https://user-images.githubusercontent.com/17328443/93847344-8c0b5d80-fc5b-11ea-9724-777b73efad45.png)

Reply:
Because *pb.User implements the fmt.Stringer interface, the string formatting provided by `Fatalf` results in output that deceptively looks identical (likely anyway). You can try dereferencing the values or using the `%#v` verb in the formatting as demoed here: https://play.golang.org/p/UkGo7MVxbRQ